### PR TITLE
Added BL to like a comment

### PR DESCRIPTION
**Pull request by RomanDeveloperAcc**
Added a functionality to the like button on comment
Link: https://github.com/ita-social-projects/GreenCity/issues/766

### CODE REVIEW

File: src/app/component/comments/components/comment-body/comment-body.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div class="comment-body-wrapper" *ngFor="let element of elements">
+<div class="comment-body-wrapper" *ngFor="let element of elements; index as i">
```
Comment (KimAT25) [inline src/app/component/comments/components/comment-body/comment-body.component.html:1]: Please fix alignment;
Check if you use "index as i";

File: src/app/component/comments/components/likes-counter/likes-counter.component.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  constructor(private commentsService: CommentsService) { }
+
+  ngOnInit() {
+    this.getLikesFromServer();
+  }
+
+  private getLikesFromServer(): void {
+    this.commentsService.getCommentLikes(this.commentId)
+      .subscribe(this.setLikes.bind(this));
```
Comment (KimAT25) [inline src/app/component/comments/components/likes-counter/likes-counter.component.ts:22]: Please change this method to not use bind();

File: src/app/component/comments/components/likes-counter/likes-counter.component.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  private setLikes(data): void {
+    console.log(data);
```
Comment (KimAT25) [inline src/app/component/comments/components/likes-counter/likes-counter.component.ts:26]: please delete all console.log()

File: src/app/component/comments/components/comment-body/comment-body.component.ts [hunk 4]
```diff
@@ -1,14 +1,31 @@
 import { Component, OnInit, Input } from '@angular/core';
 import { CommentsService } from '../../services/comments.service';
+import { UserOwnAuthService } from '@global-service/auth/user-own-auth.service';
+import set = Reflect.set;
```
Comment (KimAT25) [inline src/app/component/comments/components/comment-body/comment-body.component.ts:4]: Do you really use it ? If not please remove it.

File: src/app/component/comments/components/likes-counter/likes-counter.component.ts (hunk 3 above)
Comment (RomanDeveloperAcc) [reply src/app/component/comments/components/likes-counter/likes-counter.component.ts]: > please delete all console.log()

Fixed

File: src/app/component/comments/components/comment-body/comment-body.component.ts (hunk 4 above)
Comment (RomanDeveloperAcc) [reply src/app/component/comments/components/comment-body/comment-body.component.ts]: > Do you really use it ? If not please remove it.

Fixed

File: src/app/component/comments/components/likes-counter/likes-counter.component.ts (hunk 2 above)
Comment (RomanDeveloperAcc) [reply src/app/component/comments/components/likes-counter/likes-counter.component.ts:22]: > Please change this method to not use bind();

Won't work without bind

Comment (KimAT25) [reply src/app/component/comments/components/likes-counter/likes-counter.component.ts:22]: Everything work as before
image
image

Comment (RomanDeveloperAcc) [reply src/app/component/comments/components/likes-counter/likes-counter.component.ts:22]: > Everything work as before
> image
> image

Well, after adding types something may not work, that is why the binding is used here
works

File: src/app/component/comments/components/comment-body/comment-body.component.html (hunk 1 above)
Comment (RomanDeveloperAcc) [reply src/app/component/comments/components/comment-body/comment-body.component.html]: > Please fix alignment;
> Check if you use "index as i";

Fixed

File: src/app/component/comments/components/comment-body/comment-body.component.html [hunk 5]
```diff
@@ -9,17 +11,21 @@
...
   </div>
   <div class="comment-main-text">
     <p class="comment-text"> {{ element.text }}</p>
   </div>
   <div class="comments-elements">
-    <app-comment-like-reply></app-comment-like-reply>
-    <app-edit-comment></app-edit-comment>
-    <app-delete-comment></app-delete-comment>
+    <div class="like-reply">
+      <app-like-comment *ngIf="isLoggedIn && !(element.author.id == userId)"
```
Comment (mplesha) [inline src/app/component/comments/components/comment-body/comment-body.component.html:21]: rewrite condition

File: src/app/component/comments/components/like-comment/like-comment.component.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    this.setStartingElements(this.likeState);
+  }
+
+  private setStartingElements(state: boolean) {
+    if (state) {
+      this.like.nativeElement.srcset = this.commentsImages.liked;
+      this.span.nativeElement.innerText = 'Liked';
+    } else {
+      this.like.nativeElement.srcset = this.commentsImages.like;
+      this.span.nativeElement.innerText = 'Like';
```
Comment (mplesha) [inline src/app/component/comments/components/like-comment/like-comment.component.ts:33]: change on 2 ternary operators, like:
this.span.nativeElement.innerText = state ? 'Liked' : 'Like';

File: src/app/component/comments/components/like-comment/like-comment.component.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      });
+  }
+
+  public changeLkeBtn(): void {
+    if (this.like.nativeElement.srcset === this.commentsImages.like) {
+      this.like.nativeElement.srcset = this.commentsImages.liked;
+      this.span.nativeElement.innerText = 'Liked';
+    } else {
+      this.like.nativeElement.srcset = this.commentsImages.like;
+      this.span.nativeElement.innerText = 'Like';
```
Comment (mplesha) [inline src/app/component/comments/components/like-comment/like-comment.component.ts:50]: the same as previous

File: src/app/component/comments/components/likes-counter/likes-counter.component.ts (hunk 2 above)
Comment (KimAT25) [reply src/app/component/comments/components/likes-counter/likes-counter.component.ts]: Please add types.

File: src/app/component/comments/components/likes-counter/likes-counter.component.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+  private getLikesFromServer(): void {
+    this.commentsService.getCommentLikes(this.commentId)
+      .subscribe(data => this.setLikes(data));
+  }
+
+  private setLikes(data): void {
+    this.likes = data;
+  }
+}
```
Comment (mplesha) [inline src/app/component/comments/components/likes-counter/likes-counter.component.ts:28]: Is there any reason to use especially function <code>setLikes</code> has one line - assignment statement? Moreover, it is called once.

File: src/app/component/comments/components/reply-comment/reply-comment.component.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+  public commentsImages = {
+    reply: 'assets/img/comments/reply.png'
+  };
+
+  constructor() { }
+
+  ngOnInit() {
+  }
+
+}
```
Comment (mplesha) [inline src/app/component/comments/components/reply-comment/reply-comment.component.ts:18]: What for empty methods?

**Pull request merged**